src/org/parosproxy/paros/Constant.java

```diff
@@ -1086,23 +1088,27 @@ public static Path getPathDefaultConfigFile() {
     }
 
 	public static File getContextsDir () {
-		File f = new File(Constant.getZapHome(), USER_CONTEXTS_DIR);
-		if (! f.exists()) {
-			f.mkdirs();
-		}
-		if (f.isDirectory() && f.canWrite()) {
-			return f;
-		}
-		return Model.getSingleton().getOptionsParam().getUserDirectory();
+		return getFromHomeDir(USER_CONTEXTS_DIR);
 	}
 
 	public static File getPoliciesDir () {
-		File f = new File(Constant.getZapHome(), USER_POLICIES_DIR);
-		if (! f.exists()) {
-			f.mkdirs();
-		}
-		if (f.isDirectory() && f.canWrite()) {
-			return f;
+		return getFromHomeDir(USER_POLICIES_DIR);
+	}
+
+	private static File getFromHomeDir (String subDir) {
+		Function<Path, Boolean> createDirIfNotExists = path -> {
+			try {
+				if (!Files.exists(path)) Files.createDirectory(path);

```

Comment:
While this is syntactically correct it will read more clearly with curly braces and formatting like other ifs. 

Other than that it looks goid yo me.

Reply:
done! let me know!